vm/vm.go

```diff
@@ -136,7 +134,8 @@ type VM struct {
 	toEngine     chan<- common.Message
 
 	// State Sync client and AppRequest handlers
-	stateSyncClient *stateSyncerClient
+	*statesync.Client[*StatefulBlock]
+	*statesync.Server[*StatefulBlock]
```

Comment:
Prefer not embedding here... it can be annoying because now we can collide across functions/fields that the embedded types have.

Reply:
Will do